R/ard_stack_hierarchical.R

```diff
@@ -337,6 +337,18 @@ internal_stack_hierarchical <- function(
     }
   }
 
+  # only id and by variables in denominator
+  if (is.data.frame(denominator)) {
+    # keep any id columns that are actually in the denominator, and any by columns that exist in denominator
+    keep_cols <- unique(c(intersect(id, names(denominator)), intersect(by, names(denominator))))
+
+    # only trim if there is at least one column to keep; otherwise leave denominator as-is
+    if (!is_empty(keep_cols)) {
+      # drop any columns that are also in `variables` (or any other cols)
+      denominator <- denominator[ , keep_cols, drop = FALSE]
```

Comment:
Resolve styler check
```suggestion
      denominator <- denominator[, keep_cols, drop = FALSE]
```